### ISSUE TITLE
`<script setup>` variables declared with destructure

### ISSUE THREAD

**Issue by yyx990803** [johnsoncodehk/volar#24]
It seems `<script setup>` binding analysis is not accounting destructure patterns. This works:

```html
<script setup>
const link = useEditLink()
const url = link.url
const text = link.text
</script>
```

This doesn't:

```html
<script setup>
const { url, text } = useEditLink()
</script>
```

**Comment by johnsoncodehk** [comment]
Thanks for the report! Fixed in 0.15.10. And also fix the array binding case:
```vue
<script setup>
const [item1, item2] = useArr()
</script>

<template>{{ item1 }} {{ item2 }}</template>
```

**Comment by yyx990803** [comment]
That was fast! However, looking at the commit, not sure if the commit handles more complex destructure patterns:

```js
const { url, text = 1, foo: [bar, ...baz], ...qux } = useEditLink()
```

Some relevant test cases in `@vue/compiler-sfc` that might help: https://github.com/vuejs/vue-next/blob/master/packages/compiler-sfc/__tests__/compileScript.spec.ts#L585-L658

**Comment by johnsoncodehk** [comment]
I completely forgot about this...Thank you for the use case, very useful! I will fix it tomorrow.

**Comment by johnsoncodehk** [comment]
fixed in 0.15.11, and now has been support type-only declarations for `defineOptions()`.
